### PR TITLE
Add configurations for native-image build

### PR DESCRIPTION
**Pull request by kaneeldias**
## Purpose

Fixes: https://github.com/ballerina-platform/ballerina-standard-library/issues/3446

Related to: https://github.com/ballerina-platform/ballerina-standard-library/issues/3315

## Checklist
- [x] Linked to an issue
- [ ] Updated the specification
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

File: native/src/main/resources/META-INF/native-image/io.ballerina.stdlib/sql-native/proxy-config.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+[
+  {
+    "interfaces":["java.sql.Connection"]}
```
Comment (TharmiganK) [inline native/src/main/resources/META-INF/native-image/io.ballerina.stdlib/sql-native/proxy-config.json:3]: Lets format this json file

File: native/src/main/resources/META-INF/native-image/io.ballerina.stdlib/sql-native/resource-config.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
+  "resources":{
```
Comment (TharmiganK) [inline native/src/main/resources/META-INF/native-image/io.ballerina.stdlib/sql-native/resource-config.json:2]: Lets format this json file

File: native/src/main/resources/META-INF/native-image/io.ballerina.stdlib/sql-native/reflect-config.json [hunk 3]
```diff
@@ -0,0 +1,33 @@
+[
+{
+  "name":"[Lcom.zaxxer.hikari.util.ConcurrentBag$IConcurrentBagEntry;"
```
Comment (TharmiganK) [inline native/src/main/resources/META-INF/native-image/io.ballerina.stdlib/sql-native/reflect-config.json:3]: Lets format this json file

File: native/src/main/resources/META-INF/native-image/io.ballerina.stdlib/sql-native/resource-config.json [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  "includes":[
+    {
+      "pattern":"\\QMETA-INF/services/java.sql.Driver\\E"
+    }, 
+    {
+      "pattern":"\\Qorg/slf4j/impl/StaticLoggerBinder.class\\E"
```
Comment (TharmiganK) [inline native/src/main/resources/META-INF/native-image/io.ballerina.stdlib/sql-native/resource-config.json:8]: I think we can remove this resource class. Can you try building the image after removing this class?

Comment (kaneeldias) [reply native/src/main/resources/META-INF/native-image/io.ballerina.stdlib/sql-native/resource-config.json]: Fixed in [4263c4c](https://github.com/ballerina-platform/module-ballerina-sql/pull/563/commits/4263c4cf80dbb61c539e18ad8c99181bbc80ec4b)

File: native/src/main/resources/META-INF/native-image/io.ballerina.stdlib/sql-native/reflect-config.json (hunk 3 above)
Comment (kaneeldias) [reply native/src/main/resources/META-INF/native-image/io.ballerina.stdlib/sql-native/reflect-config.json]: Fixed in [7655a3a](https://github.com/ballerina-platform/module-ballerina-sql/pull/563/commits/7655a3a7e3e4de9e864c1cdc93eeedaad7952a61)

File: native/src/main/resources/META-INF/native-image/io.ballerina.stdlib/sql-native/resource-config.json (hunk 2 above)
Comment (kaneeldias) [reply native/src/main/resources/META-INF/native-image/io.ballerina.stdlib/sql-native/resource-config.json]: Fixed in [7655a3a](https://github.com/ballerina-platform/module-ballerina-sql/pull/563/commits/7655a3a7e3e4de9e864c1cdc93eeedaad7952a61)

File: native/src/main/resources/META-INF/native-image/io.ballerina.stdlib/sql-native/proxy-config.json (hunk 1 above)
Comment (kaneeldias) [reply native/src/main/resources/META-INF/native-image/io.ballerina.stdlib/sql-native/proxy-config.json]: Fixed in [7655a3a](https://github.com/ballerina-platform/module-ballerina-sql/pull/563/commits/7655a3a7e3e4de9e864c1cdc93eeedaad7952a61)

**Review by TharmiganK — APPROVED** [review]

**Pull request merged**